### PR TITLE
Check action cancelation in the deploy step

### PR DESCRIPTION
**Pull request by felipesu19**
This adds logic to deploy-pages to check if a run has been cancled instead of reporting a cryptic "deployment failed because artifact was not found." [Related card](https://github.com/github/pages-engineering/issues/944)

The monolith api we use to check for deployment status, includes a [deployment cancelled](https://github.com/github/pages-deployer/blob/main/internal/lib/reporter/deployment_status_reporter.go#L20), status. 

I'm only like 60% sure this will work, I'm not sure the status in redis gets updated by starting a new job. If it doesn't we probably need an ensure block of some kind in the action that run if the action is canceled, or to expand logic... somewhere to verify the status. There could also be an API endpoint i'm not thinking of. 

cc: @yoannchaudet  since you wanted to know where this was at

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -101,6 +115,8 @@ class Deployment {
           core.info(
             'Deployment temporarily failed, a retry will be automatically scheduled...'
           )
+        } else if (res.data.status == 'deployment_cancelled') {
+          core.info('Deployment cancelled. (Probably from another run)')
```
Comment (YiMysty) [inline src/index.js:119]: let's add `break` here as well, so it's not a dead loop :)

**Comment by YiMysty** [comment]
hmmm the `dist/index.js` change seems incorrect. 🤔  it almost completely removed it...

**Comment by YiMysty** [comment]
and this change won't work since we do not emit `cancel` status in build step :)

**Comment by yoannchaudet** [comment]
Before fully exiting, when cancellation is detected, let's make a call to `cancelHandler`. This is the old signal handler that makes an API call to let us know on the backend that the deployment has been cancelled. We still need that.

Outside of this, it looks great.

File: src/index.js (hunk 1 above)
Comment (felipesu19) [reply src/index.js:119]: Great point!

**Comment by felipesu19** [comment]
> and this change won't work since we do not emit `cancel` status in build step :) it's only get emitted when kick off `deploy-pages` here is how it get emitted
> 
> https://github.com/paper-spa/deploy-pages/blob/5d45a5284cd3b3bda0fb23ca8194bf6a3193389c/src/index.js#L162-L164

I was afraid of that. Need to find a different way to emit a cancelation event then. I don't super want to poll from the monolith, but we might need a new endpoint in the monolith that checks the current deployment id, and compares it to the ongoing deployment id or something.

**Comment by felipesu19** [comment]
https://github.com/github/github/pull/203449

**Pull request closed without merging**